addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -263,6 +259,156 @@ class DataEgressService extends Service {
     }
   }
 
+  async getEgressStore(requestContext, environmentId) {
+    const enableEgressStore = this.settings.get(settingKeys.enableEgressStore);
+    if (!enableEgressStore || enableEgressStore.toUpperCase() !== 'TRUE') {
+      throw this.boom.forbidden('Unable to list objects in egress store since this feature is disabled', true);
+    }
+    const curUser = _.get(requestContext, 'principalIdentifier.uid');
+    const egressStoreInfo = await this.getEgressStoreInfo(environmentId);
+    const isEgressStoreOwner = egressStoreInfo.createdBy === curUser;
+    if (!isAdmin(requestContext) && !isEgressStoreOwner) {
+      throw this.boom.forbidden(
+        `You are not authorized to perform egress store list. Please contact your administrator for more information.`,
+        true,
+      );
+    }
+    const s3Service = await this.service('s3Service');
+    // always fetch all the objects and sort and return top 100
+    const objectList = await s3Service.listAllObjects({
+      Bucket: egressStoreInfo.s3BucketName,
+      Prefix: egressStoreInfo.s3BucketPath,
+    });
+    objectList.sort((a, b) => {
+      return new Date(a.LastModified) - new Date(b.LastModified);
+    });
+    let result = [];
+    _.forEach(objectList, obj => {
+      obj.projectId = egressStoreInfo.projectId;
+      obj.workspaceId = egressStoreInfo.workspaceId;
+      obj.Size = this.bytesToSize(obj.Size);
+      const newKey = obj.Key.split('/');
+      if (newKey[1]) {
+        obj.Key = newKey[1];
+        result.push(obj);
+      }
+    });
+    if (result.length > 100) {
+      result = result.slice(0, 100);
+    }
+
+    return { objectList: result, isAbleToSubmitEgressRequest: egressStoreInfo.isAbleToSubmitEgressRequest };
+  }
+
+  bytesToSize(bytes) {
+    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
+    if (bytes === 0) return '0 Byte';
+    const i = parseInt(Math.floor(Math.log(bytes) / Math.log(1024)), 10);
```

Comment:
Seems like there is a possibility of index out of bounds exception if you go over or equal to 1024 TB. Perhaps add PB as well and over that have a special string like `>= 1024 PB`? Or you can also have a special string `>= 1024 TB` rather than undefined.

Reply:
fixed in commit: 18639d639b979c9054fa5d24fa94b3a0bdd98f11